console-plugins/dense-mapping-constraints-plugin/src/dense-mapping-selection.cc

```diff
@@ -138,6 +138,9 @@ static void filter_candidates_randomly(
 
   // Create a vector of candidate iterators.
   const std::size_t n_candidates = candidate_pairs_ptr->size();
+  if (n_candidates < n_remaining_candidates_to_keep) {
```

Comment:
if there's less candidates than allowed we can immediately exit. Otherwise it will crash in line 158 `  std::shuffle(
      v.begin() + n_reserved_candidates, v.end(),
      std::mt19937{std::random_device{}()});`